dracut/40clevis-unlock/clevis-network-generator

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -e
+
+UNIT_DIR="$1"
+
+if cat /proc/cmdline | grep -qw "rd.clevis_unlock=network"; then
+    mkdir -p "${UNIT_DIR}/clevis-unlock.service.d"
+
+    cat >${UNIT_DIR}/clevis-unlock.service.d/network_dependency.conf <<EOF
+    # Automatically generated by clevis-network-generator
+
+    [Unit]
+    Wants=systemd-networkd.service systemd-resolved.service network-online.target
+    After=systemd-networkd.service systemd-resolved.service network-online.target
+EOF
```

Comment:
We could also use this generator to avoid having to mask `systemd-cryptsetup@rootencrypted.service`. Because generators should only write to the UNIT_DIR I guess the easiest "masking" would be to generate a drop-in like
```
ExecStart=
ExecStart=true
```

Reply:
If we mask `systemd-cryptsetup@rootencrypted.service` in the generator, using a hard-coded name `systemd-cryptsetup@rootencrypted.service`, then once a user changes their Butane from

```
luks:
- name: rootencrypted
  device: "/dev/disk/by-partlabel/root"
  ...
 ```

to

```
luks:
- name: cryptoroot
  device: "/dev/disk/by-partlabel/root"
  ...
 ```

this will stop working because `systemd-cryptsetup@cryptoroot` isn't masked. I worry that having the masking depend on a magic `luks.name` will be confusing to users. Personally, I would prefer to have a Butane config that explicitly masks the relevant cryptsetup unit than a Butane config in which there is a hidden dependency between the value of `luks.name` and whether the relevant cryptsetup unit is masked.

We could try to detect the name of the dm-crypt device that contains the filesystem labeled ROOT in the generator. I can't think of a robust way to do this though. The Ignition config contains this information, but it is deleted after the first boot and so won't be available to the generator. Otherwise, only the filesystem metadata contains label information, but we would have to decrypt the dm-crypt devices in the generator to access it.

You might know a good way to achieve the masking independently of the `luks.name`---if so, let me know and I'm happy to implement it. Otherwise, I'd vote for leaving it as is and requiring explicit masking. Happy to hear other opinions though.

Reply:
Ok, thanks for the info, if I understand the relation to the other topic, the best way would be to make clevis work with the systemd-cryptsetup generated units? We could try that by adding a `luks.uuid=` cmdline arg and not passing the custom kernel arg we defined for the unlock service added here, or?

Anyway, I'm fine with the current approach - I'll make a final test to see if we can merge it as is.

Reply:
@pothos @simoncampion please check the discussion related to the initrd size before merging it. Merging this PR might increase the size of the initrd over the 50% threshold. https://github.com/flatcar/Flatcar/issues/1381

Reply:
I am afraid we cannot merge this because it will increase the initramfs size to over 50% of the 128MB boot partition. 3850.1.0 with the changed `bootengine` has >50% use on `/boot`:

```
emergency@flatcarvm ~ $ df
Filesystem                            1K-blocks  Used Available Use% Mounted on
...
/dev/vda1                                129039 64602     64437  51% /boot
...
```

I'll rebase `bootengine` and `scripts` on the main branches and build and check again, but I wouldn't be surprised if we're still above 50%.

Is it worth the effort to try to reduce the initramfs impact of this PR to stay under 50% or should we hold off the merge until there is a more general solution to the 64MB issue?

Reply:
There is ongoing effort to reduce the size under 50%. This is our best bet for the moment: https://github.com/flatcar/scripts/pull/1734.

What is the increase of this change?

Reply:
I think with the reduction in https://github.com/flatcar/scripts/pull/1734 we should be able to both merge this PR and the systemd 255 update, and then we need to look for further reduction opportunities again.